.github/actions/windows_install/action.yml

```diff
@@ -21,30 +21,62 @@ runs:
         echo $EXE
         $BIN_DIR=(Split-Path -Parent ${EXE}) # Find folder containing c compiler
         $MINGW_DIR=(Split-Path -Parent ${BIN_DIR}) # Find the enclosing mingw directory
-        $LIBRARY_DIR="$MINGW_DIR/lib"
-        echo $LIBRARY_DIR | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
-        echo "LIBRARY_DIR=${LIBRARY_DIR}" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
+        echo $MINGW_DIR | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
+        echo "MINGW_DIR=${MINGW_DIR}" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
 
         # Add DLL paths for python imports
         $PYTHON_PREFIX=(python -c "import sys; print(sys.prefix)")
         $PYTHON_SITE_PATH="$PYTHON_PREFIX\lib\site-packages"
-        echo "import os; os.add_dll_directory(r'$LIBRARY_DIR'); os.add_dll_directory(r'$BIN_DIR')" | Out-File -FilePath $PYTHON_SITE_PATH\\dll_path.pth -Encoding ascii
+        echo "import os; os.add_dll_directory(r'$MINGW_DIR/lib'); os.add_dll_directory(r'$BIN_DIR')" | Out-File -FilePath $PYTHON_SITE_PATH\\dll_path.pth -Encoding ascii
       shell: powershell
+    - name: Restore cached Lapack
+      id: cache-lapack-restore
+      uses: actions/cache/restore@v3
+      with:
+        path: |
+          C:/mingw64/lib/cmake/lapack-3.12.0/lapack-targets.cmake
+          C:/mingw64/lib/cmake/lapack-3.12.0/lapack-targets-release.cmake
+          C:/mingw64/lib/pkgconfig/lapack.pc
+          C:/mingw64/lib/cmake/lapack-3.12.0/lapack-config.cmake
+          C:/mingw64/lib/cmake/lapack-3.12.0/lapack-config-version.cmake
+          C:/mingw64/lib/pkgconfig/blas.pc
+          C:/mingw64/lib/libblas.dll.a
+          C:/mingw64/bin/libblas.dll
+          C:/mingw64/lib/liblapack.dll.a
+          C:/mingw64/bin/liblapack.dll
+        key: ${{ runner.os }}-gcc-lapack-v3.12.1
```

Comment:
Are we using version `3.12.0` or `3.12.1`?

Reply:
I am installing v3.12.1:

https://github.com/pyccel/pyccel/blob/6425fa2d298e7fc8e1efdec1b800d9d7110f1312/.github/actions/windows_install/action.yml#L53

But I am caching the locations that CMake spat out. It looks like LAPACK forgot to update this when they created the tag. The tag was created 8th Jan: https://github.com/Reference-LAPACK/lapack/releases/tag/v3.12.1
but the version was update 12th Jan:
https://github.com/Reference-LAPACK/lapack/commit/3e242fc8d5d38f0425c6018fe88abf31e1a9e80f

Reply:
Is there a way to get these locations automatically from CMake?

Reply:
I agree